tests/test_initialization.py

```diff
@@ -1846,6 +1847,98 @@ def test_road_with_conv2d_layer(self):
             get_peft_model(model, config)
 
 
+class TestDeLoRAInitialization:
+    """Basic sanity tests for the DeLoRA tuner."""
+
+    torch_device = infer_device()
+
+    def get_model(self, bias=True):
+        class MLP(nn.Module):
+            def __init__(self, bias=True):
+                super().__init__()
+                self.lin0 = nn.Linear(10, 30, bias=bias)
+                self.lin1 = nn.Linear(30, 2, bias=bias)
+
+            def forward(self, X):
+                X = self.lin0(X)
+                X = self.lin1(X)
+                return X
+
+        return MLP(bias=bias).to(self.torch_device).eval()
+
+    @pytest.fixture
+    def data(self):
+        torch.manual_seed(0)
+        return torch.randn(4, 10, device=self.torch_device)
+
+    def test_delora_injection_keeps_output_default(self, data):
+        # With use_residual_init=True (default), initial forward should match base model
+        torch.manual_seed(0)
+        base = self.get_model()
+        y_base = base(data)
+
+        cfg = DeloraConfig(target_modules=["lin0"], r=8, lambda_=8)
+        model = get_peft_model(base, cfg)
+        y_peft = model(data)
+
+        assert torch.allclose(y_base, y_peft, atol=1e-6, rtol=1e-6)
+
+    def test_delora_param_shapes(self):
+        base = self.get_model()
+        in_f, out_f = base.lin0.in_features, base.lin0.out_features
+        r = 4
+        cfg = DeloraConfig(target_modules=["lin0"], r=r, lambda_=2)
+        model = get_peft_model(base, cfg)
+
+        layer = model.lin0  # DeloraLinear wrapper
+        assert hasattr(layer, "delora_A") and hasattr(layer, "delora_B") and hasattr(layer, "delora_lambda")
+        A = layer.delora_A["default"]
+        B = layer.delora_B["default"]
+        lambda_ = layer.delora_lambda["default"]
+        assert tuple(A.shape) == (r, in_f)
+        assert tuple(B.shape) == (out_f, r)
+        assert tuple(lambda_.shape) == (1,)
+
+    def test_disable_enable_no_change(self, data):
+        base = self.get_model()
+        cfg = DeloraConfig(target_modules=["lin0"], r=8, lambda_=8)
+        model = get_peft_model(base, cfg)
+
+        y0 = model(data)
+        model.disable_adapter_layers()
+        y1 = model(data)
+        model.enable_adapter_layers()
+        y2 = model(data)
+
+        assert torch.allclose(y0, y1, atol=1e-6, rtol=1e-6)
+        assert torch.allclose(y0, y2, atol=1e-6, rtol=1e-6)
+
+    def test_merge_and_unload_same_outputs(self, data):
+        base = self.get_model()
+        cfg = DeloraConfig(target_modules=["lin0"], r=8, lambda_=8)
+        model = get_peft_model(base, cfg)
+        y_before = model(data)
+
+        merged = model.merge_and_unload()
+        merged.eval()
+        y_after = merged(data)
+        assert torch.allclose(y_before, y_after, atol=1e-6, rtol=1e-6)
```

Comment:
I think this should already be well covered by the existing tests and can be removed here.